package/boot/uboot-rockchip/Makefile

```diff
@@ -0,0 +1,55 @@
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+include $(TOPDIR)/rules.mk
+include $(INCLUDE_DIR)/kernel.mk
+
+PKG_VERSION:=2020.01
```

Comment:
2020.04 is now available

Reply:
Indeed, but it is out of scope of this PR and AFAIK there is no such rule, that the PR needs to contain latest&greatest versions of all packages, although it would be certainly nice.

Reply:
yes, but with 2020.04, there is no need for patching so rk3328 boots fine

Reply:
This PR is about RockPro64.

Reply:
Yes, but then I'll have to modify this file with the upcoming PR for rk3328, I'm just waiting for this PR to come to a conclusion

Reply:
I will try a modern u-boot release and then update this version number.

Reply:
what about python3-dev and swig, do you already have them installed on your host, I can't build u-boot on my machine without them, or is a normal dependency for openwrt ? does the buildbot have them installed ?

Reply:
I don't know if python headers are installed, but swig is definitely not. Sunxi target has a patch for that and tegra sed's out BINMAN from Kconfig. I don't know if that's feasible with rockchip.

Reply:
Very cool! Thank you very much for that info about BINMAN. 
I've added that to rockchip and tested it. It seems to work. 

@lucize As your system didn't have python3-dev and swig, could you try again if it builds for you now?

Reply:
so I added the
```
define Build/Configure
        $(SED) '/select BINMAN/d' -i $(PKG_BUILD_DIR)/arch/arm/mach-rockchip/Kconfig
        $(call Build/Configure/U-Boot)

```
but

```
Traceback (most recent call last):
  File "scripts/dtc/pylibfdt/setup.py", line 21, in <module>
    from distutils.core import setup, Extension
ModuleNotFoundError: No module named 'distutils.core'
make[3]: Leaving directory '/home/build/source/feeds/packages/utils/pciutils'
time: package/feeds/packages/pciutils/compile#0.26#0.21#0.63
scripts/dtc/pylibfdt/Makefile:27: recipe for target 'scripts/dtc/pylibfdt/_libfdt.so' failed
make[7]: *** [scripts/dtc/pylibfdt/_libfdt.so] Error 1
scripts/Makefile.build:432: recipe for target 'scripts/dtc/pylibfdt' failed

```

Reply:
I also tried to remove select SPL_OF_LIBFDT and select TPL_OF_LIBFDT I guess I'll try to research a bit more on this

Reply:
@lucize I think we have a solution. Could you pull the git, clean and try to build again?

Reply:
I think FDT needs to be disabled everywhere at least is not needed in the boot script

```
Traceback (most recent call last):
  File "./tools/dtoc/dtoc", line 44, in <module>
    import dtb_platdata
  File "/home/build/source/build_dir/target-aarch64_cortex-a53_musl/u-boot-rock64-rk3328/u-boot-2020.04/tools/dtoc/dtb_platdata.py", line 18, in <module>
    import fdt
  File "/home/build/source/build_dir/target-aarch64_cortex-a53_musl/u-boot-rock64-rk3328/u-boot-2020.04/tools/dtoc/fdt.py", line 12, in <module>
    import libfdt
ModuleNotFoundError: No module named 'libfdt'
Traceback (most recent call last):
  File "./tools/dtoc/dtoc", line 44, in <module>
    import dtb_platdata
  File "/home/build/source/build_dir/target-aarch64_cortex-a53_musl/u-boot-rock64-rk3328/u-boot-2020.04/tools/dtoc/dtb_platdata.py", line 18, in <module>
    import fdt
  File "/home/build/source/build_dir/target-aarch64_cortex-a53_musl/u-boot-rock64-rk3328/u-boot-2020.04/tools/dtoc/fdt.py", line 12, in <module>
    import libfdt
ModuleNotFoundError: No module named 'libfdt'
scripts/Makefile.spl:324: recipe for target 'tpl/dts/dt-platdata.c' failed
make[5]: *** [tpl/dts/dt-platdata.c] Error 1
make[5]: *** Waiting for unfinished jobs....
  CC      spl/disk/part_dos.o
scripts/Makefile.spl:321: recipe for target 'include/generated/dt-structs-gen.h' failed
make[5]: *** [include/generated/dt-structs-gen.h] Error 1
Makefile:1876: recipe for target 'tpl/u-boot-tpl.bin' failed
make[4]: *** [tpl/u-boot-tpl.bin] Error 2
make[4]: *** Waiting for unfinished jobs....

```

Reply:
Sorry, I can't reproduce this with the latest debian version.
I've installed a fresh Debian 10, installed git, gawk, ncurses and build-essential and it just worked.

Are you sure you're using the latest git?

Reply:
I used 
```
define Build/Configure
        $(SED) s/subdir.*pylibfdt//g $(PKG_BUILD_DIR)/scripts/dtc/Makefile
```
but jut want to let you know that rk3399 is working ok, adding 
```
  BUILD_DEVICES:= \
    rock64
```
will need the python3-dev and swig so this can go in and just see about rk3328 later